tests/unit/python/fledge/services/core/api/test_service.py

```diff
@@ -608,43 +605,121 @@ async def mock_bad_result():
         assert 404 == resp.status
         assert '{} service does not exist.'.format(name) == resp.reason
 
+    async def test_post_install_package_from_repo_already_in_progress(self, client):
+        async def async_mock(return_value):
+            return return_value
+
+        pkg_name = 'fledge-service-notification'
+        param = {"format": "repository", "name": pkg_name}
+        payload = {"return": ["status"], "where": {"column": "action", "condition": "=", "value": "install",
+                                                   "and": {"column": "name", "condition": "=", "value": pkg_name}}}
+        select_row_resp = {'count': 1, 'rows': [{
+            "id": "c5648940-31ec-4f78-a7a5-b1707e8fe578",
+            "name": pkg_name,
+            "action": "install",
+            "status": -1,
+            "log_file_uri": ""
+        }]}
+        msg = '{} package installation already in progress'.format(pkg_name)
+        storage_client_mock = MagicMock(StorageClientAsync)
+        with patch.object(connect, 'get_storage_async', return_value=storage_client_mock):
+            with patch.object(storage_client_mock, 'query_tbl_with_payload',
+                              return_value=async_mock(select_row_resp)) as query_tbl_patch:
+                resp = await client.post('/fledge/service?action=install', data=json.dumps(param))
+                assert 429 == resp.status
+                assert msg == resp.reason
+                r = await resp.text()
+                actual = json.loads(r)
+                assert {'message': msg} == actual
+            args, kwargs = query_tbl_patch.call_args_list[0]
+            assert 'packages' == args[0]
+            assert payload == json.loads(args[1])
+
+    async def test_post_install_package_from_repo_already_installed(self, client):
+        async def async_mock(return_value):
+            return return_value
+
+        pkg_name = 'fledge-service-notification'
+        param = {"format": "repository", "name": pkg_name}
+        payload = {"return": ["status"], "where": {"column": "action", "condition": "=", "value": "install",
+                                                   "and": {"column": "name", "condition": "=", "value": pkg_name}}}
+        svc_list = ["storage", "south", "notification"]
+        msg = '{} package is already installed'.format(pkg_name)
+        storage_client_mock = MagicMock(StorageClientAsync)
+        with patch.object(connect, 'get_storage_async', return_value=storage_client_mock):
+            with patch.object(storage_client_mock, 'query_tbl_with_payload',
+                              return_value=async_mock({'count': 0, 'rows': []})) as query_tbl_patch:
+                with patch.object(service, 'get_service_installed', return_value=svc_list
+                                  ) as svc_list_patch:
+                    resp = await client.post('/fledge/service?action=install', data=json.dumps(param))
+                    assert 304 == resp.status
```

Comment:
if already installed return 400 bad request or 422 not 3xx (user agent specific status codes); same for plugins